modules/eks-managed-node-group/main.tf

```diff
@@ -24,13 +24,44 @@ module "user_data" {
   user_data_template_path    = var.user_data_template_path
 }
 
+################################################################################
+# EFA Support
+################################################################################
+
+data "aws_ec2_instance_type" "this" {
+  count = var.enable_efa_support ? 1 : 0
+
+  instance_type = local.efa_instance_type
+}
+
+locals {
+  efa_instance_type = try(element(var.instance_types, 0), "")
+  # TOTO - requires v5.38.0 https://github.com/hashicorp/terraform-provider-aws/pull/35840
```

Comment:
pending v5.38.0 release of the AWS provider https://github.com/hashicorp/terraform-provider-aws/pull/35840